### ISSUE TITLE
npm install a deprecated dependency version

### ISSUE THREAD

**Issue by DataTables** [npm/npm#18171]
#### I'm opening this issue because:

  - [x] npm is producing an incorrect install.

#### What's going wrong?

`npm install` will install deprecated versions for dependencies.

#### How can the CLI team reproduce the problem?

```
npm init --yes
npm install --save datatables.net-bs
```

This will install `datatables.net-bs` at 1.10.15, and also `datatables.net` upon which the former depends. The issue is that it installed `datatables.net` 2.1.1.

That was a tag that I added by mistake and have marked as deprecated. It should never be installed unless that version is specifically asked for.

See also [this issue](https://github.com/DataTables/DataTables/issues/955#issuecomment-322489734) where it was originally brought to my attention, and has a test repo.

### supporting information:

 - `npm -v` prints: 5.0.3. Also confirmed with 5.3.0.
 - `node -v` prints: v8.1.2
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by zkat** [comment]
This is working as intended. I've replied more concretely at https://github.com/DataTables/DataTables/issues/955.

tl;dr deprecation does not, and is not intended to, alter actual dependency resolution. All it does is print messages out. Please use `^` specifiers instead of `>=`.

**Comment by DataTables** [comment]
Thanks for the feedback - much appreciated!
